client/dashboard/app/queries/site-settings.ts

```diff
@@ -1,20 +1,27 @@
+import { queryOptions, mutationOptions } from '@tanstack/react-query';
 import { fetchSiteSettings, updateSiteSettings } from '../../data/site-settings';
 import { queryClient } from '../query-client';
 import { siteQueryFilter } from './site';
 import type { SiteSettings } from '../../data/site-settings';
 
-export const siteSettingsQuery = ( siteId: number ) => ( {
-	queryKey: [ 'site', siteId, 'settings' ],
-	queryFn: () => fetchSiteSettings( siteId ),
-} );
+export const siteSettingsQuery = ( siteId: number ) =>
+	queryOptions( {
+		queryKey: [ 'site', siteId, 'settings' ],
+		queryFn: () => fetchSiteSettings( siteId ),
+	} );
 
-export const siteSettingsMutation = ( siteId: number ) => ( {
-	mutationFn: ( data: Partial< SiteSettings > ) => updateSiteSettings( siteId, data ),
-	onSuccess: ( newData: Partial< SiteSettings > ) => {
-		queryClient.setQueryData( siteSettingsQuery( siteId ).queryKey, ( oldData: SiteSettings ) => ( {
-			...oldData,
-			...newData,
-		} ) );
-		queryClient.invalidateQueries( siteQueryFilter( siteId ) );
-	},
-} );
+export const siteSettingsMutation = ( siteId: number ) =>
+	mutationOptions( {
+		mutationFn: ( data: Partial< SiteSettings > ) => updateSiteSettings( siteId, data ),
+		onSuccess: ( newData ) => {
+			queryClient.setQueryData(
+				siteSettingsQuery( siteId ).queryKey,
+				( oldData ) =>
+					oldData && {
+						...oldData,
+						...newData,
+					}
+			);
```

Comment:
Another similar bug fix.
If site settings get mutated before they are ever been queried, then this would put a partial settings object into the query cache when it is supposed to be a complete settings object.

To fix the issue, I decided that if the settings gets mutated before ever queried, then we just won't put anything into the query cache.

Reply:
Ohhhhhh nice!!!! Though it's very unlikely to write a code that exhibits that scenario 😄